app/routes.py

```diff
@@ -1 +1,227 @@
+
+    return make_response(jsonify(task_dict), 200) 
+
+@tasks_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_model(Task,task_id)
+
+    request_body = request.get_json()
+
+    task.update(request_body) 
+    db.session.commit()
+
+    task_dict = {}
+    task_dict["task"] = {"id":task.task_id, "title": task.title,"description":task.description, "is_complete":bool(task.completed_at)}
+
+    return make_response(jsonify(task_dict), 200)
+
+
+@tasks_bp.route("/<task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    task = validate_model(Task,task_id)
+
+    db.session.delete(task)
+    db.session.commit()
+    dict = {"details":f"Task {task.task_id} \"{task.title}\" successfully deleted"}
+
+    return make_response(jsonify(dict), 200)
+
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete(task_id):
+    task = validate_model(Task,task_id)
+    task.completed_at = datetime.datetime.now()
+
+    create_slack_mssg(task) 
+
+    task_dict = {}
+    task_dict["task"] = {"id":task.task_id, "title": task.title,"description":task.description, "is_complete":bool(task.completed_at)}
+
+    db.session.commit()
+
+    return make_response(jsonify(task_dict), 200)
+
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete(task_id):
+    task = validate_model(Task,task_id)
+    task.completed_at = None
+
+    task_dict = {}
+    task_dict["task"] = {"id":task.task_id, "title": task.title,"description":task.description, "is_complete":bool(task.completed_at)}
+
+    db.session.commit()
+
+    return make_response(jsonify(task_dict), 200)
+
+##############################################
+######## ROUTE TO POST SLACK MESSAGE #########
+##############################################
+
+def create_slack_mssg(task_object):
+
+    slack_token = os.environ.get("SLACK_TOKEN")
+
+    message= f"Someone just completed the task {task_object.title}"
+    args = {"token":slack_token,
+            "channel": "task-notifications",
+            "text": message
+            }
+    response=requests.post("https://slack.com/api/chat.postMessage", data=args) 
+
+    return response.json()
+
+
+###############################################
+############# ROUTES FOR GOAL #################
+###############################################
+
+@goals_bp.route("", methods=["POST"])
+def create_goals():
+    request_body = request.get_json()
+    new_goal = Goal.from_dict_goals(request_body)
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    goal_dict = {}
+    goal_dict["goal"] = {"id":new_goal.goal_id, "title": new_goal.title}
+
+    return make_response(jsonify(goal_dict), 201)
+
+@goals_bp.route("", methods=["GET"])
+def get_all_goals():
+    goals = Goal.query.all()
+    goals_response = []
+    for goal in goals:
+        goals_response.append({"id": goal.goal_id, "title": goal.title})
+
+    return (jsonify(goals_response))
```

Comment:
I don't think you need these parentheses here. It doesn't hurt anything, but it doesn't really _do_ anything either.

If you wanted to create a tuple with only one element, you have to use a `,` not just the parentheses. So it would be `(jsonify(goals_response), )` if that _was what you needed to do, but just `jsonify(goals_response)` is fine here.